go.mod

```diff
@@ -1,9 +1,23 @@
 module github.com/whyrusleeping/cbor-gen
 
-go 1.12
+go 1.18
```

Comment:
I can undo this if needed, I just wanted to use generics but it's a few lines of code difference if you'd prefer to keep this lower.